### PR TITLE
Filter Sorbet in backtrace

### PR DESCRIPTION
**Pull request by andyw8**
Paired on with @st0012 

### Motivation

When working on Ruby LSP, test failures contain Sorbet frames in the backtrace. This adds considerable noise and is generally not helpful for understand a failure.

### Implementation

There is not much guidance in the Minitest documentation for this, so this solution is based on Minitest's default filter.

### Automated Tests

n/a

### Manual Tests

Add something to a test which triggers a error, and run the test. Ensure the backtrace filter does not include any Sorbet frames.

### CODE REVIEW

File: test/test_helper.rb [hunk 1]
```diff
@@ -13,8 +13,8 @@
 require "debug"
 require "mocha/minitest"
 
-sorbet_paths = Gem.loaded_specs["sorbet-runtime"].full_require_paths.freeze
-DEBUGGER__::CONFIG[:skip_path] = Array(DEBUGGER__::CONFIG[:skip_path]) + sorbet_paths
+SORBET_PATHS = T.let(Gem.loaded_specs["sorbet-runtime"].full_require_paths.freeze, T::Array[String])
```
Comment (andyw8) [inline test/test_helper.rb:16]: Changed to a constant that we can reference it from `BacktraceWithoutSorbetFilter`.

File: test/test_helper.rb [hunk 2]
```diff
@@ -28,3 +28,20 @@ class Test
     Minitest::Test.make_my_diffs_pretty!
   end
 end
+
+# based on https://github.com/minitest/minitest/blob/master/lib/minitest.rb
+class BacktraceWithoutSorbetFilter < Minitest::BacktraceFilter
```
Comment (andyw8) [inline test/test_helper.rb:33]: fyi @st0012 - I changed this so it inherits

**Comment by andyw8** [comment]
It'll show like this:

```
DocumentSymbolExpectationsTest#test_document_symbol__keyword_splat_nil__does_not_raise:
TypeError: Parameter 'node': Expected type YARP::ModuleNode, got type YARP::DefNode with hash 2984306925210129830
Caller: /Users/andyw8/src/github.com/Shopify/ruby-lsp/lib/ruby_lsp/event_emitter.rb:120
Definition: /Users/andyw8/src/github.com/Shopify/ruby-lsp/lib/ruby_lsp/requests/document_symbol.rb:167
    /Users/andyw8/src/github.com/Shopify/ruby-lsp/lib/ruby_lsp/event_emitter.rb:120:in `block in visit_def_node
```

File: test/test_helper.rb [hunk 3]
```diff
@@ -28,3 +28,20 @@ class Test
...
+  extend T::Sig
+  sig { override.params(bt: T.nilable(T::Array[String])).returns(T::Array[String]) }
+  def filter(bt)
```
[the remark this answers is not in the document]
Comment (andyw8) [reply test/test_helper.rb:35]: Good idea, updated.

File: test/test_helper.rb [hunk 4]
```diff
@@ -28,3 +28,15 @@ class Test
     Minitest::Test.make_my_diffs_pretty!
   end
 end
+
+class BacktraceWithoutSorbetFilter < Minitest::BacktraceFilter
+  extend T::Sig
```
Comment (vinistock) [inline test/test_helper.rb:33]:
```suggestion
  extend T::Sig

```

**Review by vinistock — APPROVED** [review]

**Comment by andyw8** [comment]
We could potentially ship this with Spoom: https://github.com/Shopify/spoom/issues/456

**Pull request merged**